server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java

```diff
@@ -64,6 +64,11 @@ public TransportResyncReplicationAction(Settings settings, TransportService tran
             writeMemoryLimits);
     }
 
+    @Override
+    protected void doExecute(Task parentTask, ResyncReplicationRequest request, ActionListener<ResyncReplicationResponse> listener) {
+        assert false : "use TransportResyncReplicationAction#sync";
```

Comment:
Should we make this the default implementation for doExecute in TransportWriteAction when `rerouteBypassed()` is true? Maybe we can then also rename that method to `supportsRerouteAction`.

Reply:
Okay. This is a little tricky because I have to expose reroute in `TransportReplicationAction`. But I did that and we can discuss.

Reply:
Eh. Actually I holding off this for the moment because it gets kind of messy. We can talk more about this. I did rename the method.